modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageRetryingInputStream.java

```diff
@@ -126,13 +154,23 @@ private InputStream openStream() throws IOException {
             } catch (RetryHelper.RetryHelperException e) {
                 throw StorageException.translateAndThrow(e);
             }
-        } catch (StorageException e) {
-            if (e.getCode() == 404) {
+        } catch (StorageException storageException) {
```

Comment:
Can we add a test like `testReadFromPositionLargerThanBlobLength` in `GoogleCloudStorageThirdPartyTests`? This means you'll need to also modify `GoogleCloudStorageHttpHandler` to send `RestStatus.REQUESTED_RANGE_NOT_SATISFIED` for invalid range offsets.

Reply:
Thank you for a hint. Since all 3 cloud blob containers handle 416 error, there is a way to generalize the test for all 3 instances of 3rd part tests